gobblin-test-harness/src/main/java/gobblin/test/execution/validator/Validator.java

```diff
@@ -0,0 +1,28 @@
+package gobblin.test.execution.validator;
+
+import java.util.Comparator;
+
+/**
+ *  An interface is for defining the validator for the test. The input can be the set of input files and the output will be be output of 
+ * 
+ * @author sveerama
+ *
+ * @param <I>
+ * @param <O>
+ */
+
+public interface Validator<I, O> extends Comparator<O>
```

Comment:
There is a method in validator of isValid which return a Boolean, which compares the output produced by the test to the expected output. 

My initial thoughts were that we would execute the test and the output the test produces will be test output, and then comparator compares the output to the expected output. I guess the validator name doesnt make sense in hindsight.